### ISSUE TITLE
run-tests.fish should exit with status != 0 if a test failed.

### ISSUE THREAD

**Issue by bpinto** [bpinto/oh-my-fish#314]
```
$ script/run-tests.fish
......................✘
Expecteda/ a.20150114_020353.bak/
to equala.20150114_020354.bak/ a/............
The command "script/run-tests.fish" exited with 0.
```

From: https://travis-ci.org/bpinto/oh-my-fish/builds/46935945

**Comment by bucaran** [comment]
Ok, I know why the tests were failing. I actually allowed this to happen as I wasn't sure how this should be handled and forgot to bring it up.

Currently, only `it_` kind of tests are allowed to fail the suite, `before_all`, etc., do not even if they fail. 

I figured why would anyone add an expect condition inside a `before_`, `after_` type of tests...

**Comment by bpinto** [comment]
Fixed by #326
